File: 50 Coding Challenge Part I/TypeScript/Challenge10.ts
// Challenge 10: Calculate the sum of number in an array of numbers.

function sumArray (arr: number[]): number{
    let sum: number = 0;

    for (let i = 0; i < arr.length; i++) {
        sum += arr[i];
    }

    return sum;
}

// Driver Method
let arr: number[] = [2, 3, -1, 5, 7, 9, 10, 15, 95];
console.log(sumArray(arr));

File: 50 Coding Challenge Part I/dart/lib/challenge10.dart
/*  
  Coding Challenge 10: Calculate the sum of numbers in an array of numbers.
  Example array: [2, 3, -1, 5, 7, 9, 10, 15, 95]
  Expected output: 145

  @author: mofirojean
*/

int sumArray(List<int> arr) {
  int sum = 0;
  for (var num in arr) {
    sum += num;
  }
  return sum;
}
